src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -209,12 +212,15 @@ internal sealed class JWTHeaderWithCertificate : JWTHeader
         {
             private string _thumbPrint;
 
-            public JWTHeaderWithCertificate(IClientAssertionCertificate credential)
+            public JWTHeaderWithCertificate(IClientAssertionCertificate credential, bool sendX5C)
                 : base(credential)
             {
                 _thumbPrint = this.Credential.Thumbprint;
                 X509CertificatePublicCertValue = null;
 
+                if (!sendX5C)
```

Comment:
Do we need the instance variable _thumbPrint? Why not just use the Property - X509CertificateThumbprint?
Change it to a {get;set;}